File: LeetCodeCPP/334.increasing-triplet-subsequence.cpp
/*
 * @lc app=leetcode id=334 lang=cpp
 *
 * [334] Increasing Triplet Subsequence
 */

#include <vector>

using namespace std;

// @lc code=start
class Solution
{
public:
    bool increasingTriplet(vector<int> &nums)
    {
        int firstMin = INT_MAX;
        int secondMin = INT_MAX;
        for (auto &num : nums)
        {
            if (num <= firstMin)
                firstMin = num;
            else if (num < secondMin)
                secondMin = num;
            else if (num > secondMin)
                return true;
        }
        return false;
    }
};
// @lc code=end

File: LeetCodePython/334.increasing-triplet-subsequence.py
#
# @lc app=leetcode id=334 lang=python3
#
# [334] Increasing Triplet Subsequence
#

from math import inf
from typing import List


# @lc code=start
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        first = inf
        second = inf
        for num in nums:
            if second < num:
                return True
            if num <= first:
                first = num
            else:
                second = num

        return False


# @lc code=end
